pkg/compliance/resolver.go

```diff
@@ -687,6 +693,19 @@ func (r *defaultResolver) resolveKubeApiserver(ctx context.Context, spec InputSp
 		spec.Version = "v1"
 	}
 
+	// podsecuritypolicies have been deprecated as part of Kubernetes v1.25
+	if spec.Kind == "podsecuritypolicies" {
+		serverVersion, err := r.resolveKubeServerVersion()
+		if err != nil {
+			return nil, fmt.Errorf("cannot resolve Kubeapiserver version: %w", err)
+		}
+		major, _ := strconv.Atoi(serverVersion.Major)
+		minor, _ := strconv.Atoi(serverVersion.Minor)
+		if major >= 1 && minor >= 25 {
+			return nil, ErrIncompatibleEnvironment
+		}
+	}
```

Comment:
This looks very ad-hoc and will require maintenance each time a new Kubernetes version deprecates a given resource.
I’m wondering if we could leverage the [discovery API](https://pkg.go.dev/k8s.io/client-go/discovery#DiscoveryClient.ServerGroupsAndResources) to get an always accurate list of supported resources, instead of hard-coding the fact that specifically `PodSecurityPolicies` have been deprecated in `1.25`.

Reply:
In our use-cases, we rely on `v1` resources, which is why I opted for an ad-hoc, but really simple solution in the first place.

I've added the use of discovery API in followup commits. I personally prefer ad-hoc and simple solution, but indeed it could prove to be more robust if other APIs start to be deprecated in the future.